### PR TITLE
Add "Deposit All" button to stash

### PR DESCRIPTION
**Pull request by cullenn**
Purposely excluding "withdraw all" because that just seems like a good way to have people accidentally die with all their money and be upset.

### CODE REVIEW

**Comment by cullenn** [comment]
That's a lot of useless commits, my bad, will make sure to rebase next PR

File: src/mods/depositAll/helper.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+function deposit() {
+	// Select normal deposit button
+	document
+		.querySelector('.slot .grey.gold:not(.js-deposit-all)')
```
Comment (timebomb0) [inline src/mods/depositAll/helper.js]: Can we make all these selectors more specific?
For example, `const $stash = getWindow(WindowNames.Stash)` at the top of the methods, then all the query selectors can be `$stash.querySelector`.

Targeting such generic-sounding selectors (especially `input.formatted, `.btn.blue`) worries me that we may eventually accidentally target the wrong element.

Same feedback throughout this PR.

File: src/mods/depositAll/helper.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+		.dispatchEvent(new Event('click'));
+
+	// Input some huge value they'll have less than
+	document.querySelector('input.formatted').value = 999999999999999;
```
Comment (timebomb0) [inline src/mods/depositAll/helper.js]: Can we only call `document.querySelector` the minimum times possible? Pretty sure it makes calls to the DOM each time we invoke the method.

Same feedback throughout the PR.

For example,
```
const $currencyInput = $stash.querySelector('input.formatted');
$currencyInput.value = 99999999;
$currencyInput.dispatchEvent...
```

File: src/mods/depositAll/index.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { makeElement } from '../../utils/misc';
+import { getWindow } from '../../utils/game';
+import { deposit } from './helper';
+
+function addDepositAllButton() {
+	const $stash = getWindow('Stash');
```
Comment (timebomb0) [inline src/mods/depositAll/index.js]: Can this be `getWindow(WindowNames.Stash)`?

File: src/mods/depositAll/index.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	// If stash is closed or deposit all button is already added, we dont need to do anything
+	if (!$stash || $stash.querySelector('.js-deposit-all')) {
+		return;
+	}
+
+	// Create deposit all button and add it to stash
+	const depositTargetBtn = document.querySelector('.slot .grey.gold');
```
Comment (timebomb0) [inline src/mods/depositAll/index.js]: Nitpick, but: Code base has been prefixing all DOM elements, i.e. objects grabbed via `.querySelector()`, with a `$`. A habit I picked up a long while ago, probably from jQuery or something.
Helps identify when you're dealing with DOM elements in your JS. I find it makes the code a little easier to read.

File: src/utils/version.js [hunk 5]
```diff
@@ -4,4 +4,4 @@
 export const BREAKING_VERSION = 1;
 
 // Used for initialization message in chat, and userscript version
-export const VERSION = '1.1.7';
+export const VERSION = '1.1.8';
```
Comment (timebomb0) [inline src/utils/version.js]: What do you think about not updating the version, and I'll update it when I'm ready for release?
One less thing for contributors to worry about.

File: src/mods/depositAll/helper.js (hunk 1 above)
Comment (cullenn) [reply src/mods/depositAll/helper.js]: Good point, fixed 'em all

File: src/mods/depositAll/helper.js (hunk 2 above)
Comment (cullenn) [reply src/mods/depositAll/helper.js]: Reduced calls to querySelector where possible

File: src/mods/depositAll/index.js (hunk 3 above)
Comment (cullenn) [reply src/mods/depositAll/index.js]: Updated ui.WindowNames with a stash key and update changes to reference it

File: src/mods/depositAll/index.js (hunk 4 above)
Comment (cullenn) [reply src/mods/depositAll/index.js]: Oh sweet, I had wondered about the $varNames but forgot to ask, updated!

File: src/utils/version.js (hunk 5 above)
Comment (cullenn) [reply src/utils/version.js]: Yeah keeping this to just you definitely makes sense lol

**Comment by cullenn** [comment]
Thanks for the feedback, addressed 'em all!

File: src/mods/depositAll/helper.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+	// Input some huge value they'll have less than
+
+	const $currencyInput = $stash.querySelector('input.formatted');
+
+	$currencyInput.value = 999999999999999;
+	$currencyInput.dispatchEvent(new Event('input'));
+
+	setTimeout(function() {
+		// Only try to deposit if the Deposit button isn't disabled after the input event, avoids chat errors for invalid amount
+		if (!$stash.querySelector('.btn.blue.disabled')) {
```
Comment (timebomb0) [inline src/mods/depositAll/helper.js]: We can lessen the `querySelector` count here by 1:
```
const $depositButton = $stash.querySelector('.btn.blue');
if (!$depositButton.classList.contains('disabled')) {
    $depositButton.dispatchEvent(new Event('click'));
}
```

File: src/mods/depositAll/helper.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+import { WindowNames } from '../../utils/ui';
+
+function deposit() {
+	const $stash = getWindow(WindowNames.stash);
+
+	// Select normal deposit button
+	$stash.querySelector('.slot .grey.gold:not(.js-deposit-all)').dispatchEvent(new Event('click'));
+
+	// Input some huge value they'll have less than
+
```
Comment (timebomb0) [inline src/mods/depositAll/helper.js]: Can remove this empty line below the comment

**Comment by cullenn** [comment]
Done & done, thanks for the catch!

**Pull request merged**